CLBlast herk/syrk fixes

CNugteren opened the pull request:
This is a first attempt to fix the issues in #1963. Although this might not solve everything yet, it does solve a bug: herk is now only called for clfoat/cdouble, otherwise it calls syrk. Also fixes a missing include and some missing error status-codes.

umar456 commented:
Hey @CNugteren, Ignore the test failures for now. we are working on moving to buildbot things aren't exactly stable. I will rerun the tests once we are ready.

umar456 commented:
@CNugteren Do you mind rebasing against master.

Also, our CI is having a hard time finding this commit. I also cannot check out this PR's reference.

CNugteren commented:
I've made a bit of mess if you look at the git commit history, but it is now based on master and the changes still look OK :slightly_smiling_face:

pavanky commented:
@CNugteren Can you rebase the changes to avoid conflicts?

CNugteren commented:
> Can you rebase the changes to avoid conflicts?

I don't think there is anything to do still, it's already in sync with master, right?

I've added a minor change to fix #1963: I updated CLBlast to 1.2.0 RC1 which includes the latest TRSM/TRSV bugfixes.

CNugteren commented:
I've just updated to the new CLBlast 1.2.0 release. All issues mentioned in #1963 are now be fixed. This PR is ready to be merged in.

CNugteren commented:
@pavanky @fzimmermann89 Any updates? I believe this PR is ready to be merged in, not sure if you are still waiting for me to do something?

9prady9 commented:
@umar456 I have tested this on my Ubuntu 16.04 and couple of threading tests are failing.

blas_opencl & dot_opencl passed fine but threading_opencl failed in couple of tests
```
[ RUN      ] Threading.FFT_ALL
/home/pradeep/gitroot/ArrayFireWorkspace/morph_fixes/test/threading.cpp:317: Failure
      Expected: true
To be equal to: isUnderTolerance
      Which is: false
Expected value=(1001, 0)	 Actual Value=(0, 0) at: 0 from thread: 140227346822912

[  FAILED  ] Threading.FFT_ALL (66 ms)
[ RUN      ] Threading.BLAS
/home/pradeep/gitroot/ArrayFireWorkspace/morph_fixes/test/testHelpers.hpp:455: Failure
      Expected: 0u
      Which is: 0
To be equal to: alloc_buffers
      Which is: 2
[  FAILED  ] Threading.BLAS (199 ms)
[ RUN      ] Threading.Sparse
/home/pradeep/gitroot/ArrayFireWorkspace/morph_fixes/test/testHelpers.hpp:455: Failure
      Expected: 0u
      Which is: 0
To be equal to: alloc_buffers
      Which is: 2
[  FAILED  ] Threading.Sparse (12613 ms)
[----------] 9 tests from Threading (26655 ms total)
```
Also, now I am seeing the following thing on Linux, not sure why. This is supposed to printed out only on Windows, we don't clear the resources due to a bug in Intel OpenCL drivers.
```
Warning:  Program terminating, but clFFT resources not freed.
Please consider explicitly calling clfftTeardown( ).
```

umar456 commented:
are these tests only failing with clblast enabled?

9prady9 commented:
@umar456 Both blas_opencl, dot_opencl & threading_opencl passed fine. This can be merged.
